File: 0017-letter-combinations-of-a-phone-number/0017-letter-combinations-of-a-phone-number.cpp
class Solution {
public:
    vector<string> numpad = {"", "", "abc", "def", "ghi", "jkl",
                                "mno", "pqrs", "tuv", "wxyz"};
    
    vector<string> ans;
    
    void help(string digits, int index, string curr){
        if(index == digits.size()){
            ans.push_back(curr);
            return;
        }
        for(char ch: numpad[digits[index] - '0']){
            help(digits, index+1, curr+ch);
        }
        
        
    }
    
    vector<string> letterCombinations(string digits) {
        if(digits.size() == 0)
            return {};
        help(digits, 0, "");  
        return ans;
    }
};

File: 0017-letter-combinations-of-a-phone-number/0017-letter-combinations-of-a-phone-number.py
class Solution:
    def letterCombinations(self, digits: str) -> List[str]:
        if(len(digits) ==0):
            return []
        numpad = ["", "", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz"]
        ans = []
        
        def help(digits, index, curr):
            if(index >= len(digits)):
                ans.append(curr)
                return
            
            for ch in numpad[ord(digits[index])-ord('0')]:
                help(digits, index+1, curr + ch)
                
        help(digits, 0, "")
        
        return ans
        
        